core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -71,15 +74,18 @@
 
     private ShardSearchFailure[] shardFailures;
 
+    private Clusters clusters;
+
     private long tookInMillis;
 
     public SearchResponse() {
     }
 
     public SearchResponse(SearchResponseSections internalResponse, String scrollId, int totalShards, int successfulShards,
-                          int skippedShards, long tookInMillis, ShardSearchFailure[] shardFailures) {
+                          int skippedShards, long tookInMillis, ShardSearchFailure[] shardFailures, @Nullable Clusters clusters) {
```

Comment:
instead of allowing null should we make it required and just don't render it if `total == successful` ? I don't like null invariants

Reply:
that's what I had before, but I think in this specific case null makes sense. Rendering only when total != successful becomes confusing as the same response is also used for scroll responses where rendering this section has no value. Also, you don't see the section unless something went wrong. I think it would be better to print the section out for all CCS requests, and never when going only local. null allows to distinguish between responses that involved CCS and responses that come from local nodes only. Would you prefer to have a placeholder instead of null, called let's say `EMPTY`, which indicates that the section has no meaningful value and should not be printed out?

Reply:
placeholder would be fine IMO